galsim/fitswcs.py

```diff
@@ -964,9 +991,12 @@ def _read_header(self, header):
         # Start by reading the basic WCS stuff that most types have.
         ctype1 = header['CTYPE1']
         ctype2 = header['CTYPE2']
-        if not (ctype1.startswith('RA---') and ctype2.startswith('DEC--')):
-            raise NotImplementedError("GSFitsWCS can only handle cases where CTYPE1 is RA " +
-                                      "and CTYPE2 is DEC")
+        if ctype1.startswith('DEC--') and ctype2.startswith('RA---'):
+            flip = True
+        elif ctype1.startswith('RA---') and ctype2.startswith('DEC--'):
+            flip = False
+        else:
+            raise RuntimeError("The WCS read in does not define a pair of celestial axes" )
```

Comment:
Might be slightly more informative to also spit out the ctype1 and ctype2 values (since they are used to determine whether this error should be raised).